### PR TITLE
Remove usage of obsolete constructors and members

### PR DESCRIPTION
**Pull request by axunonb**
* All YAXLibTests and DemoApplication updated
* Removed SerializerOptionsTests (containing no tests, but still covered 100%)

### CODE REVIEW

File: DemoApplication/FormMain.cs [hunk 1]
```diff
@@ -125,7 +126,7 @@ private void btnDeserializeFromFile_Click(object sender, EventArgs e)
             OnDeserialize(true);
         }
 
-        private void OnDeserialize(bool openFromFile)
+         private void OnDeserialize(bool openFromFile)
```
Comment (304NotModified) [inline DemoApplication/FormMain.cs:129]: nitpick, strange space?

File: YAXLibTests/SerializationTest.cs [hunk 2]
```diff
@@ -2512,11 +2726,7 @@ public void PolymorphicCollectionWithNullElements()
         [Test]
         public void SerializingNullValues()
         {
-            var book = new Book();
-            book.Price = 10;
-            book.Title = null;
```
Comment (304NotModified) [inline YAXLibTests/SerializationTest.cs:2517]: lol useful :P

File: YAXLibTests/SerializerOptionsTests.cs [hunk 3]
```diff
@@ -1,75 +0,0 @@
...
-using FluentAssertions;
-using NUnit.Framework;
-using YAXLib;
-using YAXLib.Enums;
-using YAXLib.Options;
-
-namespace YAXLibTests
-{
-    [TestFixture]
-    public class SerializerOptionsTests
```
Comment (304NotModified) [inline YAXLibTests/SerializerOptionsTests.cs:14]: why are these removed?

Comment (304NotModified) [reply YAXLibTests/SerializerOptionsTests.cs:14]: Guess not useful.

**Review by 304NotModified — APPROVED** [review]
Damn nice work! I know this is a lot of work!

File: YAXLibTests/SerializerOptionsTests.cs (hunk 3 above)
Comment (axunonb) [reply YAXLibTests/SerializerOptionsTests.cs:14]: Removed SerializerOptionsTests: containing no tests, but still covered 100% because in use ~1 million times

File: DemoApplication/FormMain.cs (hunk 1 above)
Comment (axunonb) [reply DemoApplication/FormMain.cs:129]: At last I found the blank I lost today, lol
Will fix

File: YAXLibTests/SerializationTest.cs (hunk 2 above)
Comment (axunonb) [reply YAXLibTests/SerializationTest.cs:2517]: Price can't be null at least :)
The rest of blanks went here, strange. Will fix.
